src/main/scala/xiangshan/cache/dcache/data/BankedDataArray.scala

```diff
@@ -375,13 +402,30 @@ class SramedDataArray(implicit p: Parameters) extends AbstractBankedDataArray {
   })
 
   // read conflict
-  val rr_bank_conflict = Seq.tabulate(LoadPipelineWidth)(x => Seq.tabulate(LoadPipelineWidth)(y =>
-    io.read(x).valid && io.read(y).valid &&
-    div_addrs(x) === div_addrs(y) &&
-    (io.read(x).bits.bankMask & io.read(y).bits.bankMask) =/= 0.U &&
-    io.read(x).bits.way_en === io.read(y).bits.way_en &&
-    set_addrs(x) =/= set_addrs(y)
-  ))
+  val rr_bank_conflict = Seq.tabulate(LoadPipelineWidth)(x => Seq.tabulate(LoadPipelineWidth)(y => {
+    if (x == y) {
+      false.B
+    } else {
+      io.read(x).valid && io.read(y).valid &&
+        div_addrs(x) === div_addrs(y) &&
+        (io.read(x).bits.bankMask & io.read(y).bits.bankMask) =/= 0.U &&
+        io.read(x).bits.way_en === io.read(y).bits.way_en &&
+        set_addrs(x) =/= set_addrs(y)
+    }
+  }))
+  val load_req_with_bank_conflict = rr_bank_conflict.map(_.reduce(_ || _))
+  val load_req_valid = io.read.map(_.valid)
+  val load_req_lqIdx = io.read.map(_.bits.lqIdx)
+  val load_req_index = (0 until LoadPipelineWidth).map(_.asUInt)
+
+
+  val load_req_bank_conflict_selcet = selcetOldestPort(load_req_valid, load_req_lqIdx, load_req_index)
+  val load_req_bank_select_port  = UIntToOH(load_req_bank_conflict_selcet._2).asBools
+
+  val rr_bank_conflict_oldest = (0 until LoadPipelineWidth).map(i =>
+    !load_req_bank_select_port(i) && load_req_with_bank_conflict(i)
```

Comment:
In this case, there will be multiple replay: ldu0,1,2's lqptr= [5,7,6], bank_conflict only in ldu1 and ldu2. Ideally only replay ldu1, but here both ldu1 and ldu2 will replay.

Reply:
> In this case, there will be multiple replay: ldu0,1,2's lqptr= [5,7,6], bank_conflict only in ldu1 and ldu2. Ideally only replay ldu1, but here both ldu1 and ldu2 will replay.

Yes, this is a problem that I will fix